File: SFML-PONG/src/Ball.cpp
#include "../include/Ball.h"

sf::Vector2<float> BallSpeed(0.3, 0.3);

Ball::Ball()
{
}

Ball::Ball(float RADIUS, sf::Color COLOR)
{
    Object.setRadius(RADIUS);
    Object.setFillColor(COLOR);

}

void Ball::Init(sf::Vector2f POSITION)
{
    Object.setPosition(POSITION);
    currentSpeed = BallSpeed;
}

void Ball::deflectX()
{
    currentSpeed.x = -currentSpeed.x;
}

void Ball::deflectY()
{
    currentSpeed.y = -currentSpeed.y;
}

void Ball::accelerate(Paddle PLAYER)
{
    currentSpeed.y = (Object.getGlobalBounds().top
                      + Object.getGlobalBounds().height / 2
                      - PLAYER.getObject().getGlobalBounds().top
                      - PLAYER.getObject().getGlobalBounds().height / 2) / 100;
}

void Ball::stopMoving()
{
    currentSpeed = sf::Vector2<float>(0.0f, 0.0f);
}
void Ball::resetSpeed(int sign)
{
    currentSpeed = BallSpeed;
    BallSpeed.y = BallSpeed.y * sign;
    BallSpeed.x = BallSpeed.x * sign;
}

bool Ball::touches(sf::RectangleShape RECTANGLE)
{
    if(Object.getGlobalBounds().intersects(RECTANGLE.getGlobalBounds()))
        return true;
    return false;
}

bool Ball::touches(sf::FloatRect RECT)
{
    if(Object.getGlobalBounds().intersects(RECT))
        return true;
    return false;
}

sf::Vector2f Ball::getBallPosition()
{
    return Object.getPosition();
}

float Ball::getBallRadius()
{
    return Object.getRadius();
}

void Ball::move()
{
    Object.move(currentSpeed.x, currentSpeed.y);
}

sf::CircleShape Ball::getObject()
{
    return Object;
}

Ball::~Ball()
{
}

File: JSFML-PONG/src/pong/Ball.java
package pong;

import org.jsfml.graphics.*;
import org.jsfml.system.*;

public class Ball {

	private CircleShape Object = new CircleShape();
	Vector2f currentSpeed = new Vector2f(0.1f, 0.1f);
	Vector2f BallSpeed = new Vector2f(0.3f, 0.3f);

	public Ball(float RADIUS) {
		Object.setRadius(RADIUS);
		Object.setFillColor(Color.WHITE);
	};

	void Init(Vector2f POSITION) {
		Object.setPosition(POSITION);
		currentSpeed = BallSpeed;
	};

	void deflectX() {
		currentSpeed = new Vector2f(-currentSpeed.x, currentSpeed.y);
	};

	void deflectY() {
		currentSpeed = new Vector2f(currentSpeed.x, -currentSpeed.y);
	};

	// move the ball
	void move() {
		Object.move(currentSpeed.x, currentSpeed.y);
	};

	void accelerate(Paddle PLAYER) {
		float a = (Object.getGlobalBounds().top
				+ (Object.getGlobalBounds().height / 2)
				- PLAYER.getObject().getGlobalBounds().top - (PLAYER
				.getObject().getGlobalBounds().height / 2)) / 100f;
		currentSpeed = new Vector2f(currentSpeed.x, a);
	};

	void stopMoving() {
		currentSpeed = new Vector2f(0.0f, 0.0f);
	};

	void resetSpeed(int sign) {
		currentSpeed = BallSpeed;
		BallSpeed = new Vector2f(BallSpeed.x, BallSpeed.y * sign);
		BallSpeed = new Vector2f(BallSpeed.x * sign, BallSpeed.y);
	};

	boolean touches(RectangleShape RECTANGLE) {
		if (Object.getGlobalBounds().intersection(RECTANGLE.getGlobalBounds()) != null)
			return true;
		return false;
	}

	Vector2f getBallPosition() {
		return Object.getPosition();
	}

	float getBallRadius() {
		return Object.getRadius();
	}

	CircleShape getObject() {
		return Object;
	}

}
